Automattic/wp-calypso
# Email stats loading states

#### Proposed Changes

This PR does a couple of things to make loading states work in Email open stats:

- Changes the shape of the stats.emails.requests store to: `[siteId][postId][period][statType][date]` (adding `date`). Because the endpoint only returns one entry (for the current period/date) for the fields `client`, `device`, `country`), we need this additional key to be able to determine if a request is currently in-flight.
- In `isLoadingTabs` first check if we have the needed data available in the Redux store, return true if so
- For `<StatEmailModule>` a new selector is introduced called `shouldShowLoadingIndicator`, this also first checks if the needed data is available in Redux. 
- Finally, in `getEmailStatsNormalizedData`change order of the parameters, to reflect the shape of the Redux state (cleanup)

#### Testing Instructions

* Visit `/stats/email/open/<site>/day/<postid>?flags=newsletter/stats`. You can add the startDate parameter, to load the stats for a specific date.
* When selecting another date you should see a loading indicator for panels where the data is not yet fetched
* Once the data is fetched for a particular date is fetched, you should be able to click around and the data should be presented instantly


https://user-images.githubusercontent.com/528287/210817696-aa70aee0-f3b1-4e07-8056-0b1c65bd909d.mp4

(It will look a bit different, because since starting this PR lots of changes have landed).

You might want to clear your local persisted state before testing this by running this in your JavaScript console. (Or clicking through Devtools: Application > IndexedDb).

```js
var req = indexedDB.deleteDatabase('calypso');
req.onsuccess = function () {
    console.log("Deleted database successfully");
};
```


#### Pre-merge Checklist





Related to #
